File: maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        result =  nums[0]
        windowSum, windowStart =  nums[0],  nums[0]
        for windowEnd in range(1, len(nums)):
            windowSum += nums[windowEnd]
            windowSum = max(windowSum,  nums[windowEnd])
            result = max(result, windowSum )
            
        return result

File: maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int largest = nums.at(0);
        int curr = 0;
        
        for (auto num: nums){
            if (curr < 0)
                curr = 0;
            curr += num;
            largest = max(curr, largest);
        }
        return largest;
    }
};